Specification.md

```diff
@@ -1,17 +1,145 @@
-## Gas limit
+# Specification
 
-By default `web3.js` uses `web3.eth.Contract.method.myMethod().estimateGas()`, but this value can differ from the 
-actual gas that will be used, if the contract you are calling depends on the blockhash, blocknumber or any other source of 
-randomness making the gas cost nondeterministic.
+The CLI should package several "core" extensions.
+The API these extensions expose is aimed at `node` environments.
+The aim is to provide convenience to power-users & devs.
 
-E.g.: the [`MiniMeToken`](https://github.com/aragon/aragon-apps/blob/master/shared/minime/contracts/MiniMeToken.sol) contract which snapshots balances at certain block numbers.
+## IPFS `@aragon/ipfs-utils`
 
-Considering the above-mentioned behavior, the CLI should calculate the recommended gas limit as follows:
-- `recommendedGas = estimatedGas`, if `estimatedGas > upperGasLimit` or
+Goals:
+
+- installing & starting a daemon with the right configuration
+- interacting with local/remote nodes: authentication, configuration, pinning, etc.
+- help setup a "production" node (?)
+- automatically pin anything published to an APM repository (?)
```

Comment:
I wouldn't really consider these last two part of the flow; for production you're likely to set up your own infra and the last item is very abusable (we actually used to do this but have since moved away to curating the pinning set).